benchmarks/benchmarks/bench_ufunc.py

```diff
@@ -60,6 +60,31 @@ def time_ufunc_types(self, ufuncname):
         [self.f(*arg) for arg in self.args]
 
 
+class UFuncBenchmark(Benchmark):
+    params = ['bool', 'uint8', 'int8', 'int32', 'uint64', 'int64', 'float16',
+              'float32', 'float64', 'complex128', 'datetime64[ns]',
+              'timedelta64[ns]']
+    param_names = ['dtype']
+
+    def setup(self, dtype):
+        self.arr = np.full(100000, 1, dtype=dtype)
+        self.val = self.arr[0]
+
+
+class MinMax(UFuncBenchmark):
+    def time_fmax(self, dtype):
+        np.fmax(self.arr, self.arr)
```

Comment:
Having spent too much understanding benchmarks, I would suggest preallocating, and prefaulting your output arrays.